src/v/kafka/server/connection_context.h

```diff
@@ -191,8 +191,16 @@ class connection_context final
     ss::future<> handle_auth_v0(size_t);
 
 private:
+    /**
+     * Bundles together a response and its associated resources.
+     */
+    struct response_and_resources {
+        response_ptr response;
+        session_resources resources;
+    };
+
     using sequence_id = named_type<uint64_t, struct kafka_protocol_sequence>;
-    using map_t = absl::flat_hash_map<sequence_id, response_ptr>;
+    using map_t = absl::flat_hash_map<sequence_id, response_and_resources>;
```

Comment:
should we consider using node_hash_map now, as currently we actually allocate session resources in the map, (previously it was only a pointer) ?

Reply:
Just to be clear, `response_and_resources` is just two pointers:

```
struct response_and_resources {
        response_ptr response;
        session_resources::pointer resources;
    };
```

Total size is 24 bytes. The old size was 16 bytes, so this is only a "modest" (50%) increase in size.

That said, I'll go with your judgement here if you think it should be changed.